cc-cloudfuse

```diff
@@ -7,13 +7,24 @@ fi
 shift
 ARGS=$@
 
-# check auth url
-if ! [ -z ${OS_AUTH_URL:+x} ]; then
-    if [[ $OS_AUTH_URL != *?v3 ]]; then
-        OS_AUTH_URL="$OS_AUTH_URL/v3"
-    fi
+OPENSTACK_VENDOR_DATA='http://169.254.169.254/openstack/latest/vendor_data.json'
+OPENSTACK_VENDOR_DATA_2='http://169.254.169.254/openstack/latest/vendor_data2.json'
+
+JSON_VENDOR_DATA=$(curl -s $OPENSTACK_VENDOR_DATA_2)
+if [ "$JSON_VENDOR_DATA" = '{}' ]; then
+  JSON_VENDOR_DATA=$(curl -s $OPENSTACK_VENDOR_DATA)
 fi
 
+function extract_json_key {
+  RESULT=$(echo "$2" | sed "s/.*$1\": \"//g" | sed 's/".*//g')
+  echo "$RESULT"
+}
+
+OS_TOKEN=$(extract_json_key "instance_metrics_writer_token" "$JSON_VENDOR_DATA")
+OS_PROJECT_ID=$(extract_json_key "project_id" "$JSON_VENDOR_DATA")
```

Comment:
This is one of my pet peeves with Keystone. It doesn't understand the token is already scoped to a project! Lame.